collectors/charts.d.plugin/speedtest/speedtest.chart.sh

```diff
@@ -0,0 +1,160 @@
+DIMENSION speedtest.idle_latency '' absolute 1 1
+CHART speedtest.download_latency '' 'Download Latency' 'milliseconds' 'latency' 'speedtest.download_latency' line $((speedtest_priority)) $speedtest_update_every '' '' 'speedtest'
+DIMENSION speedtest.download_latency '' absolute 1 1
+CHART speedtest.upload_latency '' 'Upload Latency' 'milliseconds' 'latency' 'speedtest.upload_latency' line $((speedtest_priority)) $speedtest_update_every '' '' 'speedtest'
+DIMENSION speedtest.upload_latency '' absolute 1 1
+CHART speedtest.idle_jitter '' 'Idle Jitter' 'milliseconds' 'jitter' 'speedtest.idle_jitter' line $((speedtest_priority)) $speedtest_update_every '' '' 'speedtest'
+DIMENSION speedtest.idle_jitter '' absolute 1 1
+CHART speedtest.download_jitter '' 'Download Jitter' 'milliseconds' 'jitter' 'speedtest.download_jitter' line $((speedtest_priority)) $speedtest_update_every '' '' 'speedtest'
+DIMENSION speedtest.download_jitter '' absolute 1 1
+CHART speedtest.upload_jitter '' 'Upload Jitter' 'milliseconds' 'jitter' 'speedtest.upload_jitter' line $((speedtest_priority)) $speedtest_update_every '' '' 'speedtest'
+DIMENSION speedtest.upload_jitter '' absolute 1 1
+CHART speedtest.download_bytes '' 'Bytes downloaded' 'bytes' 'bandwidth' 'speedtest.download_bytes' line $((speedtest_priority)) $speedtest_update_every '' '' 'speedtest'
+DIMENSION speedtest.download_bytes '' absolute 1 1
+CHART speedtest.upload_bytes '' 'Bytes uploaded' 'bytes' 'bandwidth' 'speedtest.upload_bytes' line $((speedtest_priority)) $speedtest_update_every '' '' 'speedtest'
+DIMENSION speedtest.upload_bytes '' absolute 1 1
+EOF
+
+  return 0
+}
+
+# _update is called continuously, to collect the values
+speedtest_update() {
+  # the first argument to this function is the microseconds since last update
+  # pass this parameter to the BEGIN statement (see below).
+
+  speedtest_get || return 1
+
+  # write the result of the work.
+  cat << VALUESEOF
+BEGIN speedtest.download $1
+SET speedtest.download = $speedtest_download
+END
+BEGIN speedtest.upload $1
+SET speedtest.upload = $speedtest_upload
+END
+BEGIN speedtest.packetloss $1
+SET speedtest.packetloss = $speedtest_packetloss
+END
+BEGIN speedtest.idle_latency $1
+SET speedtest.idle_latency = $speedtest_idle_latency
+END
+BEGIN speedtest.download_latency $1
+SET speedtest.download_latency = $speedtest_download_latency
+END
+BEGIN speedtest.upload_latency $1
+SET speedtest.upload_latency = $speedtest_upload_latency
+END
+BEGIN speedtest.idle_jitter $1
+SET speedtest.idle_jitter = $speedtest_idle_jitter
+END
+BEGIN speedtest.download_jitter $1
+SET speedtest.download_jitter = $speedtest_download_jitter
+END
+BEGIN speedtest.upload_jitter $1
+SET speedtest.upload_jitter = $speedtest_upload_jitter
+END
+BEGIN speedtest.download_bytes $1
+SET speedtest.download_bytes = $speedtest_download_bytes
+END
+BEGIN speedtest.upload_bytes $1
+SET speedtest.upload_bytes = $speedtest_upload_bytes
+END
+VALUESEOF
+
+  return 0
+}
```

Comment:
To finish this review step, I ran `shellcheck` on this file and I got these warnings:

```sh
bash-5.2$ shellcheck speedtest.chart.sh 

In a.sh line 23:
speedtest_magic_number=
^--------------------^ SC2034: speedtest_magic_number appears unused. Verify use (or export if used externally).


In a.sh line 38:
speedtest_last=0
^------------^ SC2034: speedtest_last appears unused. Verify use (or export if used externally).


In a.sh line 39:
speedtest_count=0
^-------------^ SC2034: speedtest_count appears unused. Verify use (or export if used externally).


In a.sh line 53:
  speedtest_download=$(echo $output | awk -F, '{print $6}' | tr -d '"')
                            ^-----^ SC2086: Double quote to prevent globbing and word splitting.


In a.sh line 54:
  speedtest_upload=$(echo $output | awk -F, '{print $7}' | tr -d '"')
                          ^-----^ SC2086: Double quote to prevent globbing and word splitting.


In a.sh line 55:
  speedtest_download_bytes=$(echo $output | awk -F, '{print $8}' | tr -d '"')
                                  ^-----^ SC2086: Double quote to prevent globbing and word splitting.


In a.sh line 56:
  speedtest_upload_bytes=$(echo $output | awk -F, '{print $9}' | tr -d '"')
                                ^-----^ SC2086: Double quote to prevent globbing and word splitting.


In a.sh line 57:
  speedtest_idle_latency=$(echo $output | awk -F, '{print $3}' | tr -d '"')
                                ^-----^ SC2086: Double quote to prevent globbing and word splitting.


In a.sh line 58:
  speedtest_download_latency=$(echo $output | awk -F, '{print $12}' | tr -d '"')
                                    ^-----^ SC2086: Double quote to prevent globbing and word splitting.


In a.sh line 59:
  speedtest_upload_latency=$(echo $output | awk -F, '{print $16}' | tr -d '"')
                                  ^-----^ SC2086: Double quote to prevent globbing and word splitting.


In a.sh line 60:
  speedtest_idle_jitter=$(echo $output | awk -F, '{print $4}' | tr -d '"')
                               ^-----^ SC2086: Double quote to prevent globbing and word splitting.


In a.sh line 61:
  speedtest_download_jitter=$(echo $output | awk -F, '{print $13}' | tr -d '"')
                                   ^-----^ SC2086: Double quote to prevent globbing and word splitting.


In a.sh line 62:
  speedtest_upload_jitter=$(echo $output | awk -F, '{print $17}' | tr -d '"')
                                 ^-----^ SC2086: Double quote to prevent globbing and word splitting.


In a.sh line 63:
  speedtest_packetloss=$(echo $output | awk -F, '{print $5}' | tr -d '"')
                              ^-----^ SC2086: Double quote to prevent globbing and word splitting.

```

Please, let us work to reduce `shellcheck` warnings to zero :smiley: .

Reply:
Will do!

Reply:
Do you use VS Code? If so, install https://marketplace.visualstudio.com/items?itemName=timonwong.shellcheck